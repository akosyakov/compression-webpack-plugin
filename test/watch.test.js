import MemoryFileSystem from 'memory-fs';
import webpack from 'webpack';

import Plugin from '../src/index';

import { getAssetsInfo } from './helpers';

const fs = require('fs');

describe('when in watch mode', () => {
  let compiler;

  let originalContent;

  beforeEach(() => {
    originalContent = fs.readFileSync(`${__dirname}/fixtures/async.js`);
    compiler = webpack({
      entry: {
        js: `${__dirname}/fixtures/entry.js`,
      },
      output: {
        path: `${__dirname}/dist`,
        filename: '[name].js',
      },
      mode: 'development',
      module: {
        rules: [
          {
            test: /\.(png|jpg|gif|svg)$/i,
            use: [
              {
                loader: 'file-loader',
              },
            ],
          },
        ],
      },
    });
    compiler.outputFileSystem = new MemoryFileSystem();
  });

  afterEach(() => {
    fs.writeFileSync(`${__dirname}/fixtures/async.js`, originalContent);
  });

  it('matches shansphot', (done) => {
    new Plugin().apply(compiler);

    let prepare = true;

    const watcher = compiler.watch({}, (err, stats) => {
      if (err) done(err);

      expect(getAssetsInfo(stats.compilation.assets)).toMatchSnapshot('assets');

      if (prepare) {
        prepare = false;
        fs.writeFileSync(
          `${__dirname}/fixtures/async.js`,
          'console.log("Hello World")'
        );
      } else {
        watcher.close();
        done();
      }
    });
  });
});
