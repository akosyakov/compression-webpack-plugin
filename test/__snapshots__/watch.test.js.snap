// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when in watch mode matches shansphot: assets 1`] = `
Array [
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg.gz",
    393,
  ],
  Array [
    "async.js",
    368,
  ],
  Array [
    "async.js.gz",
    212,
  ],
  Array [
    "js.js",
    11602,
  ],
  Array [
    "js.js.gz",
    3005,
  ],
]
`;

exports[`when in watch mode matches shansphot: assets 2`] = `
Array [
  Array [
    "5b1f36bc41ab31f5b801d48ba1d65781.png",
    78117,
  ],
  Array [
    "96621d3c37d96ad3bf792fcc848d912f.svg",
    672,
  ],
  Array [
    "async.js",
    396,
  ],
  Array [
    "async.js.gz",
    235,
  ],
  Array [
    "js.js",
    11602,
  ],
]
`;
